src/components/wrapper/index.js

```diff
@@ -12,26 +13,32 @@ import './wrapper.css';
 /**
  * Main app container. Handles showing/hiding the sidebar nav, and theme classes.
  */
-export const Wrapper = ({ loading, theme }) => (
-  <div
-    className={classnames('kedro-pipeline', {
-      'kui-theme--dark': theme === 'dark',
-      'kui-theme--light': theme === 'light'
-    })}>
-    <h1 className="pipeline-title">Kedro-Viz</h1>
-    <Sidebar />
-    <MetaData />
-    <div className="pipeline-wrapper">
-      <FlowChart />
-      <LoadingIcon className="pipeline-wrapper__loading" visible={loading} />
+export const Wrapper = ({ loading, theme, isChonky, displayChonkyGraph }) => {
+  const displayWarning = isChonky && !displayChonkyGraph ? true : false;
+
+  return (
+    <div
+      className={classnames('kedro-pipeline', {
+        'kui-theme--dark': theme === 'dark',
+        'kui-theme--light': theme === 'light'
+      })}>
+      <h1 className="pipeline-title">Kedro-Viz</h1>
+      <Sidebar />
+      <MetaData />
+      <div className="pipeline-wrapper">
+        {displayWarning === true ? <Modal /> : <FlowChart />}
```

Comment:
Thanks for the recommendation!  I completely understand your point about the thoughts on this as an improvement on performance - I'm just not entirely sure whether this will is necessary given the following reasons: 

1. The graphState will still need to be recalculated on every new selection of the tags, which by then the react component will need to be re-rendered anyways. 
2. Rendering the flowchart component in the background at this stage might be an overkill as the user will likely have moved on with a selection within seconds of seeing this warning ( there isn't any point in rendering an empty graph in the background ) 
3.  During our previous discussion, it was suggested that this warning to be displayed in the fastest way on first load, which is best achieved by minimising the amount of elements to be rendered on first load. This is hence my earlier refactor to move the trigger of this warning up in `calculateGraph`, along with the need to set up a series of new actions just for this - to only load this warning, which is a simple div as compared to a complex flowchart component. 
4. I had tried testing the performance by toggling the tag selection back and forth, and for me there has been no huge impact on performance. 

My pragmatic view is that, given that I don't see any huge performance issues, I am sure whether there are much value for this refactor in enhancing the performance ( or whether it's necessary for a simple warning that would not be used 99% of the time. )

Reply:
Ah I see your points, but I actually refer more to a general browser best practice worth keeping an eye on.

From a performance perspective it's best to mount components only once and never unmount them where you know they will always be in use, as to avoid DOM thrashing and other initialisation / disposal costs each time (which in turn cause the UI to freeze user input and render jank).

Since we know the flowchart will always be needed and is a core component, it's a good idea to eager mount it in this case and never unmount it, even if it is visibly empty. Maybe the flowchart component needs a little adapting to allow this?

Reply:
yes indeed, the flowchart component do need adapting for this case - I had just tried a quick try / implementation of rendering the flowchart while the warning message is in place ( i.e the toggle graph action along with other actions for graph information calculation has not been dispatched, meaning invalid inputs for the flowChart), and it is returning a series of console errors ( see below). 
![image](https://user-images.githubusercontent.com/27775658/107069712-43071080-67da-11eb-8443-e16d1ec43484.png)

In the meantime while clicking through the different tags to switch between the warning and the rendering of the flowchart for the selected tags, I don't notice any difference in the speed / performance in the display of the graph ( i.e it still takes a long time to load the newly calculated graph given it still needs to go through `updateGraph` calculations.) 

I completely understand your view on browser DOM best practices, however with all the potential errors thrown ( from the invalid data and further errors thrown each time the user scrolls on the warning message triggered by the flowchart component), furthermore without any particular observed improvements,  it doesn't seem to justify the need to render the flowchart component at the same time. ( of course, if this do get raised by any user in future observations, we can always revisit this again.)

Reply:
@bru5 I've given this a thought over the weekend, as how you have mentioned before, there might be a better and less intrusive solution to the large pipeline, such as going in the other direction to automatically only render a subset of the nodes and adding a reminder to the user to change their filters on the left if they want to render more nodes, etc. In the meantime, this feature is likely just a temporary solution while we think of implementing a better more intuitive solution to solve this.

Given this, the effort to refactor flowchart ( along with the tests and any potential other tests this breaks) to accomodate this might be an overkill at this stage. However, if you feel strongly about this change, we can file a new ticket to tackle this. Sounds good?

Reply:
Ah okay I wasn't expecting it would require a lot of changes, I guess since the message is to be only shown once per session at the start then this can probably be revisited later. For anything regular though definitely better to use a CSS based overlay / toggle.

Reply:
This is now resolved with the latest refactor